airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -0,0 +1,321 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import csv
+import json
+import time
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from requests import codes, exceptions
+
+from .api import UNSUPPORTED_FILTERING_STREAMS, Salesforce
+from .rate_limiting import default_backoff_handler
+
+
+class SalesforceStream(HttpStream, ABC):
+
+    version = "v52.0"
+    limit = 2000
+
+    def __init__(self, sf_api: Salesforce, pk: str, stream_name: str, schema: dict = None, **kwargs):
+        super().__init__(**kwargs)
+        self.sf_api = sf_api
+        self.pk = pk
+        self.stream_name = stream_name
+        self.schema = schema
+
+    @property
+    def name(self) -> str:
+        return self.stream_name
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self.pk
+
+    @property
+    def url_base(self) -> str:
+        return self.sf_api.instance_url
+
+    def path(self, **kwargs) -> str:
+        return f"/services/data/{self.version}/queryAll"
+
+    def next_page_token(self, response: requests.Response) -> str:
+        response_data = response.json()
+        if len(response_data["records"]) == self.limit and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+            return f"WHERE {self.primary_key} > '{response_data['records'][-1][self.primary_key]}' "
+
+    def request_params(
```

Comment:
can we add a link to the salesforce query syntax here in the function doc string?